java/client/build.gradle

```diff
@@ -134,26 +124,83 @@ tasks.register('testFfi', Test) {
 }
 
 tasks.register('buildAll') {
-    dependsOn 'protobuf', 'buildRust', 'testFfi'
+    dependsOn 'protobuf', 'buildRustRelease', 'testFfi'
     finalizedBy 'build'
 }
 
 compileJava.dependsOn('protobuf')
 clean.dependsOn('cleanProtobuf', 'cleanRust')
-test.dependsOn('buildRust')
-testFfi.dependsOn('buildRust')
+
+tasks.register('copyNativeLib', Copy) {
+    from "${projectDir}/../target/release"
+    include "*.dylib", "*.so"
+    into sourceSets.main.output.resourcesDir
+}
+
+jar.dependsOn('copyNativeLib')
+copyNativeLib.dependsOn('buildRustRelease')
+compileTestJava.dependsOn('publishToMavenLocal')
 
 test {
     exclude "glide/ffi/FfiTest.class"
 }
 
+sourceSets {
+    main {
+        java {
+            srcDir 'src/main/java'
+        }
+        resources {
+            srcDir 'src/main/resources'
+        }
+    }
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            from components.java
+            pom {
+                name = 'glide-for-valkey'
+                description = 'General Language Independent Driver for the Enterprise (GLIDE) for Valkey'
+                url = 'https://github.com/aws/glide-for-redis.git'
+                inceptionYear = '2024'
+                scm {
+                    url = 'https://github.com/aws/glide-for-redis/tree/main'
+                    connection = 'scm:git:ssh://git@github.com/aws/glide-for-redis.git'
+                    developerConnection = 'scm:git:ssh://git@github.com/aws/glide-for-redis.git'
+                }
+                licenses {
+                    license {
+                        name = 'The Apache License, Version 2.0'
+                        url = 'https://www.apache.org/licenses/LICENSE-2.0.txt'
+                        distribution = 'repo'
+                    }
+                }
+                developers {
+                    developer {
+                        name = 'glide-for-valkey'
+                        url = 'https://github.com/aws/glide-for-redis.git'
+                    }
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            url = 'https://aws.oss.sonatype.org/service/local/staging/deploy/maven2/'
+            credentials(PasswordCredentials)
```

Comment:
We dont have this yet, may be worth placing a TODO
But... we can try copy that from another repo on aws, e.g.
https://github.com/aws/aws-sdk-java-v2/blob/019159d4bd88b7739e122189bec469cc39ea8bd5/buildspecs/resources/release-settings.xml#L4

Reply:
I found better example https://github.com/aws/amazon-neptune-jdbc-driver/blob/develop/build.gradle